src/filerepository.js

```diff
@@ -120,32 +128,26 @@ export default class FileRepository extends Plugin {
 	}
 
 	/**
-	 * Returns the loader associated with specified file.
+	 * Returns the loader associated with specified file or promise.
 	 *
 	 * To get loader by id use `fileRepository.loaders.get( id )`.
 	 *
-	 * @param {File} file Native file handle.
+	 * @param {File|Promise} fileOrPromise Native file or promise handle.
```

Comment:
As Maciek wrote - you can use `Promise.<File>` to represent this. Also please check other occurrences of `Promise` in the docs to change them to `Promise.<File>`.